### PR TITLE
Add canvas image content support

### PR DESCRIPTION
**Pull request by wuzekang**

### CODE REVIEW

File: src/snapshot.ts [hunk 1]
```diff
@@ -193,6 +193,10 @@ function serializeNode(
           attributes.selected = (n as HTMLOptionElement).selected;
         }
       }
+      // canvas image data
+      if (tagName === 'canvas') {
+        attributes._dataURL = (n as HTMLCanvasElement).toDataURL()
```
Comment (Yuyz0112) [inline src/snapshot.ts]: use `rr_dataURL` as the key so we know its a rrweb-added attribute.

**Comment by Yuyz0112** [comment]
Thanks a lot!

**Comment by oasif** [comment]
I tried to get it to work but the canvas does not render. Will you be able to provide full canvas support please?

**Comment by wuzekang** [comment]
I checked, snapshot to rebuild is no problem. 
https://codesandbox.io/s/reverent-grass-ci9w7
@oasif

**Comment by oasif** [comment]
@wuzekang, static canvas works well. thank you! However canvas mutations are not detected.  If it is not possible to detect mutation on a canvas, is it possible to force snapshot on a canvas on an interval?

**Comment by vipulwairagade** [comment]
@Yuyz0112 @wuzekang @oasif , I want to record mutations done on canvas and I have multiple canvas as I am rendering a pdf document in the web. Please if you can help me with it.

[replay_2020_05_07T14_06_47.zip](https://github.com/rrweb-io/rrweb-snapshot/files/4595589/replay_2020_05_07T14_06_47.zip)
As you can see I am able to record other parts but not the content of pages and mutations done.

**Pull request merged**